src/operator/controllers/certificates/spirecertgen/spire_certificate_generator.go

```diff
@@ -0,0 +1,73 @@
+package spirecertgen
+
+import (
+	"context"
+	"fmt"
+	"github.com/otterize/spire-integration-operator/src/controllers/secrets/types"
+	"github.com/otterize/spire-integration-operator/src/controllers/spireclient/bundles"
+	"github.com/otterize/spire-integration-operator/src/controllers/spireclient/svids"
+	"time"
+)
+
+type SpireCertificateDataGenerator struct {
+	bundlesStore bundles.Store
+	svidsStore   svids.Store
+}
+
+func NewSpireCertificateDataGenerator(bundlesStore bundles.Store, svidsStore svids.Store) *SpireCertificateDataGenerator {
+	return &SpireCertificateDataGenerator{bundlesStore: bundlesStore, svidsStore: svidsStore}
+}
+
+func (m *SpireCertificateDataGenerator) Generate(ctx context.Context, config secretstypes.SecretConfig) (secretstypes.CertificateData, error) {
+	trustBundle, err := m.bundlesStore.GetTrustBundle(ctx)
+	if err != nil {
+		return secretstypes.CertificateData{}, err
+	}
+
+	privateKey, err := m.svidsStore.GeneratePrivateKey()
+
+	if err != nil {
+		return secretstypes.CertificateData{}, err
+	}
+
+	svid, err := m.svidsStore.GetX509SVID(ctx, config.EntryID, privateKey)
+	if err != nil {
+		return secretstypes.CertificateData{}, err
+	}
+
+	expiry := time.Unix(svid.ExpiresAt, 0)
+	expiryStr := expiry.Format(time.RFC3339)
+
+	secretData, err := m.generateSecretData(trustBundle, svid, config.CertConfig)
+	if err != nil {
+		return secretstypes.CertificateData{}, nil
+	}
+	return secretstypes.CertificateData{Files: secretData, ExpiryStr: expiryStr}, nil
+}
+
+func (m *SpireCertificateDataGenerator) generateSecretData(trustBundle bundles.EncodedTrustBundle, svid svids.EncodedX509SVID, certConfig secretstypes.CertConfig) (map[string][]byte, error) {
+	switch certConfig.CertType {
```

Comment:
It looks like this function returns a map, with two possible "formats" (keys), and those two formats are fixed.

You should split this function into two functions, that return a struct (instead of a function) with the appropriate fields.